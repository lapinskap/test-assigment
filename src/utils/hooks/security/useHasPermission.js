import { useEffect } from 'react';
import RbsContext from '../../RoleBasedSecurity/RbsContext';
import { godPermission } from '../../RoleBasedSecurity/permissions';

const { useState, useContext } = require('react');

const useHasPermission = (aclKey) => {
  const { userInfo } = useContext(RbsContext);
  const [result, setResult] = useState(null);
  useEffect(() => {
    // const permissions = userInfo.getPermissions();
    // setResult(Boolean(permissions.includes(godPermission) || permissions.includes(aclKey)));
    setResult(true);
  }, [userInfo, aclKey]);

  return result;
};

export default useHasPermission;
