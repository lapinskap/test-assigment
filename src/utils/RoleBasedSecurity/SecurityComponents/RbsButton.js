import PropTypes from 'prop-types';
import { Button } from 'reactstrap';
import React from 'react';
import __ from '../../Translations';
import useHasPermission from '../../hooks/security/useHasPermission';

const RbsButton = ({
  permission, children, className, disabled, title, ...props
}) => {
  let buttonDisabled = disabled;
  let buttonClassName = buttonDisabled ? `${className} enable-pointer-event` : className;
  let buttonTitle = title;
  const hasPermissions = useHasPermission(permission);
  if (permission && !hasPermissions) {
    buttonDisabled = true;
    buttonTitle = __('Nie masz uprawnie≈Ñ ({0})', [permission]);
    buttonClassName = buttonClassName ? `${buttonClassName} not-allowed` : 'not-allowed';
  }

  return (
  // eslint-disable-next-line react/jsx-props-no-spreading
    <Button {...props} disabled={buttonDisabled} className={buttonClassName} title={buttonTitle}>
      {children}
    </Button>
  );
};

export default RbsButton;

RbsButton.propTypes = {
  className: PropTypes.string,
  permission: PropTypes.string,
  title: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

RbsButton.defaultProps = {
  permission: '',
  title: null,
  className: null,
  disabled: false,
};
