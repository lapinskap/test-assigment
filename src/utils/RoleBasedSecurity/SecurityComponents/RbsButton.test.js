import React from 'react';
import renderer from 'react-test-renderer';
import RbsButton from './RbsButton';
import { getPermissionsWrapper } from '../../hooks/security/useHasPermission.test';

describe('<RbsButton />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when have permissions ', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);

    const tree = renderer
      .create(
        <Wrapper>
          <RbsButton
            permission="test1:company:company"
          />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when have not permissions ', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);

    const tree = renderer
      .create(
        <Wrapper>
          <RbsButton
            permission="random:scope"
          />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when have permissions and button has props', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);

    const tree = renderer
      .create(
        <Wrapper>
          <RbsButton
            permission="random:scope"
            className="classRandom"
          />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
