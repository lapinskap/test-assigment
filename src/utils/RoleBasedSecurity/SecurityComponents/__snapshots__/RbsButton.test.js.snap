// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RbsButton /> renders correctly when have not permissions  1`] = `
<button
  aria-label={null}
  className="not-allowed btn btn-secondary disabled"
  disabled={true}
  onClick={[Function]}
  title="Nie masz uprawnień (random:scope)"
/>
`;

exports[`<RbsButton /> renders correctly when have permissions  1`] = `
<button
  aria-label={null}
  className="not-allowed btn btn-secondary disabled"
  disabled={true}
  onClick={[Function]}
  title="Nie masz uprawnień (test1:company:company)"
/>
`;

exports[`<RbsButton /> renders correctly when have permissions and button has props 1`] = `
<button
  aria-label={null}
  className="classRandom not-allowed btn btn-secondary disabled"
  disabled={true}
  onClick={[Function]}
  title="Nie masz uprawnień (random:scope)"
/>
`;
